### PR TITLE
Prometheus measurements can return violations errors conditionally.

### PR DESCRIPTION
**Pull request by oxddr**
Also create override file which enable violations in APIResponsivenessPrometheus. They are disable for now. 

refs https://github.com/kubernetes/perf-tests/issues/498

/assign @krzysied @mm4tt

### CODE REVIEW

**Comment by oxddr** [comment]
Let's wait for tests to pass.
/hold

File: clusterloader2/pkg/measurement/common/slos/prometheus_measurement.go [hunk 1]
```diff
@@ -72,12 +72,21 @@ func (m *prometheusMeasurement) Execute(config *measurement.MeasurementConfig) (
...
 		c := config.PrometheusFramework.GetClientSets().GetClient()
 		executor := measurementutil.NewQueryExecutor(c)
 
 		summary, err := m.gatherer.Gather(executor, m.startTime)
-		if err != nil && !errors.IsMetricViolationError(err) {
-			return nil, err
+		if err != nil {
+			if !errors.IsMetricViolationError(err) {
+				return nil, err
+			} else if !enableViolations {
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/slos/prometheus_measurement.go]: nit: remove else (unnecessary nesting).

**Comment by mm4tt** [comment]
/lgtm
/approve

Nice!

**Comment by oxddr** [comment]
/retest
prow hasn't picked up fix yet

**Pull request merged**